### PR TITLE
Use devcontainer to run some CI tests

### PR DESCRIPTION
**Pull request by vncoelho**
Try to close https://github.com/neo-project/neo/issues/3319

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
       uses: actions/checkout@v4
 
     - name: Build and run dev container task
-      uses: devcontainers/ci@v0.4
+      uses: devcontainers/ci@v0.3
```
Comment (cschuchardt88) [inline .github/workflows/main.yml:116]: Maybe use [v0.3.1900000349](https://github.com/devcontainers/ci/releases/tag/v0.3.1900000349)?

Comment (vncoelho) [reply .github/workflows/main.yml:116]: I saw, that, but what should be the nomenclature there? all these numbers?

**Comment by vncoelho** [comment]
Now the trick...Reverting commit 2bd2a4522f365fb314f9b49aeef866b60da986c6 and check if it will fail

**Comment by vncoelho** [comment]
Did not fail, publish command is missing

**Comment by vncoelho** [comment]
I am still not able to reproduce the local environment correctly

File: .github/workflows/main.yml (hunk 1 above)
Comment (cschuchardt88) [reply .github/workflows/main.yml:116]: https://github.com/devcontainers/ci/tree/chrmarti/federated-ids?tab=readme-ov-file#changelog

Comment (cschuchardt88) [reply .github/workflows/main.yml:116]: Also I have an image already. https://github.com/cschuchardt88/neo/pkgs/container/neo-cli

**Pull request closed without merging**